packages/ketcher-core/src/application/render/raphaelRender.ts

```diff
@@ -0,0 +1,279 @@
+/****************************************************************************
+ * Copyright 2021 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+import { Box2Abs, Struct, Vec2 } from 'domain/entities'
+
+import Raphael from './raphael-ext'
+import { ReStruct } from './restruct'
+import { Scale } from 'domain/helpers'
+import defaultOptions from './options'
+import draw from './draw'
+import { RaphaelPaper } from 'raphael'
+// import { RenderOptions } from './render.types'
+
+function calcExtend(
+  scaledSz: Vec2,
+  x0: number,
+  y0: number,
+  x1: number,
+  y1: number
+) {
+  // eslint-disable-line max-params
+  let ex = x0 < 0 ? -x0 : 0
+  let ey = y0 < 0 ? -y0 : 0
+
+  if (scaledSz.x < x1) ex += x1 - scaledSz.x
+  if (scaledSz.y < y1) ey += y1 - scaledSz.y
+  return new Vec2(ex, ey)
+}
+
+class Render {
+  public clientArea: HTMLElement
+  private userOpts: any
+  public paper: RaphaelPaper
+  private sz: Vec2
+  public ctab: ReStruct
+  public options: any
+  private renderWidth: number
+  private renderHeight: number
+  private useOldZoom: boolean
+  private oldCb: Box2Abs | null
```

Comment:
Not pushing, but if you know what sz, ctab, oldCb mean, would be great to name them something meaningful 😄